File: settlement/room/service/module/RoomServiceNeed.java
package settlement.room.service.module;

import init.type.NEEDS;
import settlement.room.main.RoomBlueprintImp;
import settlement.room.main.util.RoomInitData;

public abstract class RoomServiceNeed extends RoomServiceAccess {

	public RoomServiceNeed(RoomBlueprintImp b, RoomInitData data) {
		super(b, data, NEEDS.MAP().read(data.data().json("SERVICE")));
	}
	
	public interface ROOM_SERVICE_NEED_HASER extends ROOM_SERVICE_ACCESS_HASER{
		
		@Override
		RoomServiceNeed service();

	}

}

File: LegacySource/settlement/room/service/module/RoomServiceNeed.cs
using settlement.room.main;
using settlement.room.main.util;
using init.type;

namespace settlement.room.service.module
{
    public abstract class RoomServiceNeed : RoomServiceAccess
    {
        public RoomServiceNeed(RoomBlueprintImp b, RoomInitData data)
            : base(b, data, NEEDS.MAP().Read(data.data().json("SERVICE")))
        {
        }

        public interface ROOM_SERVICE_NEED_HASER : ROOM_SERVICE_ACCESS_HASER
        {
            new RoomServiceNeed service();
        }
    }
}